### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by christophwelling**
With the new changes to the event structure and the detector description, some of the information in the NuRadioReco paper is pretty outdated, so we should explain this in the documentation (and try to keep it up to date).
I started with an explanation of the new event structure, but feel free to expand this with anything that you feel is missing.

### CODE REVIEW

File: NuRadioReco/autodoc/pages/detector.rst [hunk 1]
```diff
@@ -1,2 +1,145 @@
...
+-----------------------------
+The ``Detector`` class provides easy access to information about a given detector.
+Instead of manually sending requests to an SQL database or parsing through
+JSON files, a ``Detector`` object is created and given the appropriate SQL
+configuration or JSON detector description and information about the detector
+is requested via *get*-methods. By setting the ``Detector`` to a specific time
+via the ``update`` function, it provides the detector configuration as it was
+at  specific time.
+
+.. Important:: Since the detector configuration can change with time, the detector has to be set to the correct time using the ``update`` method.
```
Comment (cg-laser) [inline NuRadioReco/autodoc/pages/detector.rst]: Can you add small code snippets to the documentation? Here e.g. something like
```
from NuRadioReco.detector import detector
from astropy.time import Time # not sure if this is the correct syntac
det = detector.Detector()
det.update(Time.now())
```

**Comment by cg-laser** [comment]
@christophwelling awesome that you take the lead in adding this long-term documentation! This helps a lot.
I made few improvements to the detector description. 

I think it would be very helpful if you could add code snippets to illustrate the usage.

**Comment by christophwelling** [comment]
I added some code snippets to the detector documentation.
I also started deploying this decumentation online. Though this is still a pull request, I don't see any harm in letting everyone see the latest documentation

**Comment by anelles** [comment]
I would suggest to mark this ready for review and just have another pull request with more documentation, if needed. Just to keep the master up to date quickly.

**Comment by christophwelling** [comment]
It only changes the documentation (most of the commits are there because I branched this off another branch that has been merged by now), which shouldn't be affected by anything else. So I'm not worried about any merge conflicts.
Also, deployment of the documentation is completely independent from merging this. In fact, the online documentation is already the latest version on this branch.
I can merge if you prefer, but I see no reason to hurry.

**Comment by christophwelling** [comment]
Okay, this should do it for now. But feel free to add stuff if you think I missed something.

**Pull request merged**